app/controllers/likes_controller.rb

```diff
@@ -29,6 +29,6 @@ def destroy
   private
 
   def extract_likeable
-    params[:likeable].singularize.classify.constantize
+    params[:likeable_type].singularize.classify.constantize
```

Comment:
If you leave off constantize you should be able to pull out the class type when you search for id params in your controller as well.  Another option